docs: made a separate page for namespaces (#473)
elastisys/compliantkubernetes: docs/user-guide/namespaces.md

---
description: How to work with namespaces in Elastisys Compliant Kubernetes, the security-focused Kubernetes distribution.
tags:

> - ISO 27001 A.12.1.4

Reviewer: ```suggestion
- ISO 27001 A.12.1.4 Separation of Development, Testing & Operational Environments
```

> - ISO 27001 A.14.2.5

Reviewer: ```suggestion
- ISO 27001 A.14.2.5 Secure System Engineering Principles
```

---

# Namespaces

## HNC

> [Hierarchical Namespace Controller](https://github.com/kubernetes-sigs/hierarchical-namespaces) (HNC) is included in Compliant Kubernetes. It allows the super application developer to manage namespaces as subnamespaces and delegates access automatically. From the perspective of Kubernetes these are regular namespaces, but these can be modified via a namespaced resource by the user. Building a good namespace structure will let you be able to apply namespace-scoped RBAC resources to multiple namespaces at once.

Reviewer 2: ```suggestion
[Hierarchical Namespace Controller](https://github.com/kubernetes-sigs/hierarchical-namespaces) (HNC) is included in Compliant Kubernetes. It allows the super application developer to manage namespaces as subnamespaces and delegates access automatically. From the perspective of Kubernetes these are regular namespaces, but these can be modified via a namespaced resource by the user. Building a good namespace structure will enable you to apply namespace-scoped RBAC resources to multiple namespaces at once.
```
